test/common/upstream/health_checker_impl_test.cc

```diff
@@ -848,6 +894,57 @@ TEST_F(TcpHealthCheckerImplTest, Timeout) {
   cluster_->runCallbacks({}, removed);
 }
 
+TEST_F(TcpHealthCheckerImplTest, TimeoutWithReuseConnection) {
```

Comment:
Nit: Generally prefer a one liner about `TEST_F` describing what the test does in plain language. I know we're not very consistent about this.